cpp/src/arrow/array/builder_dict.cc

```diff
@@ -258,13 +258,28 @@ Status DictionaryBuilder<T>::AppendNull() {
   return values_builder_.AppendNull();
 }
 
+template <typename T>
+Status DictionaryBuilder<T>::AppendNulls(int64_t length) {
+  length_ += length;
+  null_count_ += length;
+
+  return values_builder_.AppendNulls(length);
+}
+
 Status DictionaryBuilder<NullType>::AppendNull() {
   length_ += 1;
   null_count_ += 1;
 
   return values_builder_.AppendNull();
 }
 
+Status DictionaryBuilder<NullType>::AppendNulls(int64_t length) {
```

Comment:
Do you need to specialize for NullType if the body is the same?

Reply:
Unfortunately yes, because those are separate implementations (otherwise the name is not found when linking).